amp/WEB-INF/src/org/digijava/module/aim/util/ActivityVersionUtil.java

```diff
@@ -346,13 +346,19 @@ public static AmpActivityVersion initializeActivity(AmpActivityVersion act) thro
         return act;
     }
     
-    public static Map<Long, Map<String, List<CompareOutput>>> compareActivities(Long[] ActivitiesId) throws Exception {
+    public static Map<Long, Map<String, List<CompareOutput>>> compareActivities(ArrayList<Long> ActivitiesId) throws Exception {
         
-        Session session = PersistenceManager.getCurrentSession();
         Map<Long, Map<String, List<CompareOutput>>> activitiesCollection = new HashMap<Long, Map<String, List<CompareOutput>>>();
-        for (int i=0; i < ActivitiesId.length; i++) {
-            Map<String, List<CompareOutput>> ActivityList = compareActivities(ActivitiesId[i]);
-            activitiesCollection.put(ActivitiesId[i], ActivityList);
+        for (int i=0; i < ActivitiesId.size(); i++) {
+            Session session = PersistenceManager.getCurrentSession();
```

Comment:
I missed this in my previous review you already have the method that compares one activity with its previous version, we shall reuse that so we dont have duplicate code